lib/api/model/model.dart

```diff
@@ -370,6 +388,24 @@ class ZulipStream {
     _$ZulipStreamFromJson(json);
 
   Map<String, dynamic> toJson() => _$ZulipStreamToJson(this);
+
+  bool hasPostingPermission(User user,
+      {required DateTime byDate, required int realmWaitingPeriodThreshold}) {
+    final role = user.role;
+    // We let the users with [unknown] role to send the message, then the server
+    // will decide to accept it or not based on its actual role.
+    if (role == UserRole.unknown) return true;
```

Comment:
Cool, this solution looks good. Thanks for the revision!

The PR is still in maintainer review, so I'll leave @PIG208 to do the next round of review on the PR as a whole.